File: fooddairy-quarkus/src/main/java/org/acme/model/Ingredient.java
package org.acme.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Ingredient extends PanacheEntityBase {

    @GeneratedValue
    @Id
    private int ingredientId;

    private String name;
    private String unit;
    private int calories;
    private float fat;
    private float carbohydrates;
    private float sugar;
    private float protein;
    private float salt;

    public Ingredient(){

    }

    public Ingredient(String name, String unit, int calories, float fat, float carbohydrates, float sugar, float protein, float salt) {
        this.name = name;
        this.unit = unit;
        this.calories = calories;
        this.fat = fat;
        this.carbohydrates = carbohydrates;
        this.sugar = sugar;
        this.protein = protein;
        this.salt = salt;
    }

    public int getIngredientId() {
        return ingredientId;
    }

    public void setIngredientId(int ingredientId) {
        this.ingredientId = ingredientId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    public int getCalories() {
        return calories;
    }

    public void setCalories(int calories) {
        this.calories = calories;
    }

    public float getFat() {
        return fat;
    }

    public void setFat(float fat) {
        this.fat = fat;
    }

    public float getCarbohydrates() {
        return carbohydrates;
    }

    public void setCarbohydrates(float carbohydrates) {
        this.carbohydrates = carbohydrates;
    }

    public float getSugar() {
        return sugar;
    }

    public void setSugar(float sugar) {
        this.sugar = sugar;
    }

    public float getProtein() {
        return protein;
    }

    public void setProtein(float protein) {
        this.protein = protein;
    }

    public float getSalt() {
        return salt;
    }

    public void setSalt(float salt) {
        this.salt = salt;
    }
}

File: FoodDairy-Project/FoodDairy/app/src/main/java/com/example/fooddairy/db/model/Ingredient.kt
package com.example.fooddairy.db.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Ingredient(

    @PrimaryKey(autoGenerate = true)
    //@ColumnInfo(name = "ingredient_id")
    var ingredientId: Int,

    @ColumnInfo(name = "ingredient_name")
    var name: String,

    @ColumnInfo(name = "ingredient_calories")
    var calories: Int = 0,
    var unit: String = "",
    var fat: Float = 0f,
    var carbohydrates: Float = 0f,
    var sugar: Float = 0f,
    var protein: Float = 0f,
    var salt: Float = 0f){

    fun getNutritionFactString(): String{
        var result: String = ""
        val pad: Int = 30
        result += "Nutrition facts per 100 $unit:\n"
        result += ("Calories".padEnd(pad, ' ')+"$calories\n")
        result += ("Fat".padEnd(pad, ' ')+"$fat\n")
        result += ("Carbohydrates".padEnd(pad, ' ')+"$carbohydrates\n")
        result += ("Protein".padEnd(pad, ' ')+"$protein\n")
        result += ("Sugar".padEnd(pad, ' ')+"$sugar\n")
        result += ("Salt".padEnd(pad, ' ')+"$salt")
        return result
    }

    override fun toString(): String {
        return name
    }
}